sql/table.h

```diff
@@ -2922,6 +2923,51 @@ inline void mark_as_null_row(TABLE *table)
 
 bool is_simple_order(ORDER *order);
 
+class Open_tables_backup;
+class TR_table: public TABLE_LIST
```

Comment:
I don't think inherit from `TABLE_LIST` is a good idea here. It's definitely incorrect from logical point of view. I think your need here a `TABLE_LIST` list class member.

Reply:
I don't see what is incorrect. What is the point?

Reply:
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c120-use-class-hierarchies-to-represent-concepts-with-inherent-hierarchical-structure-only
> Do not use inheritance when simply having a data member will do.

Reply:
I don't like dogmas from some large and wise documents. This is not always good in practice.

Reply:
There are no bugs. So it's ok to just leave it as it.